src/s2e/Plugins/Lua/LuaInstructionAnnotation.cpp

```diff
@@ -177,6 +179,29 @@ void LuaInstructionAnnotation::onInstruction(S2EExecutionState *state, uint64_t
     if (luaAnnotation.exitCpuLoop()) {
         throw CpuExitException();
     }
+
+    if (luaAnnotation.doSkip()) {
+        TranslationBlock *tb = state->getTb();
+        int i;
+        uint64_t next_pc = 0;
+
+        getDebugStream(state) << "Annotation " << it->annotationName << " asked to skip instruction at "
+                              << hexval(it->pc) << '\n';
+
+        // Get address of next instruction (= current PC + size of current instruction)
+        for (i = 0; i < tb->precise_entries && next_pc == 0; i++) {
+            if (tb->precise_pcs[i].guest_pc_increment == pc - tb->pc) {
```

Comment:
I think tb->pc + tb->precise_pcs[i].guest_pc_increment == pc would make it easier to understand.